### ISSUE TITLE
`GetChatHistory` in a few threads

### ISSUE THREAD

**Issue by akulik512** [tdlib/td#2582]

**Comment by akulik512** [comment]
Hello @levlam 
I accidentally posted a question without a description. Fixed

**Comment by levlam** [comment]
No, you can't do this. You can't send the next request before receiving response to the previous one, because you don't know the next from_message_id.

**Comment by akulik512** [comment]
Before `GetChatHistory` I send `SearchPublicChat` which has the last message published to the channel and contains`messageId` (something like this). This id will be equal to the number of all messages in the channel. So if the first message's batch has 100 items then the next `messageId` will be 200. I think we can calculate the next `from_message_id`.

But now I see another problem it seems like I can't stop consuming after getting 100 messages or only get 100 messages. I just can point out where should we start (`from_message_id`) but it will return all other channel messages and can't be stopped.

**Comment by akulik512** [comment]
Oh no, I can stop the process, I see it, sorry :smile: 
Ok, can we move on to the initial question? Will we get a ban if we send a few `GetChatHistory` requests at the same time using one session?

**Comment by levlam** [comment]
> This id will be equal to the number of all messages in the channel.

It is not, it is just an identifier. There is no way to "calculate" the next `from_message_id` in advance.

And yes, the account can be banned for Telegram API abuse.
